indexer/src/main.rs

```diff
@@ -121,24 +129,68 @@ async fn main() -> anyhow::Result<()> {
         )
     })?;
 
-    let mut sink: CardanoSink = match config.sink {
-        SinkConfig::Cardano { .. } => CardanoSink::new(config.sink, exec_plan).await?,
-        _ => todo!("not supported yet"),
+    let (network, mut sink) = match config.sink {
```

Comment:
putting out the network is not generic enough, the code that is setting up source and sink shouldn't know anything how the configs and the internals work